src/PostgREST/Error.hs

```diff
@@ -474,7 +474,7 @@ pgErrorStatus authed (SQL.SessionUsageError (SQL.QueryError _ _ (SQL.ResultError
         '3':'9':_ -> HTTP.status500 -- external routine invocation
         '3':'B':_ -> HTTP.status500 -- savepoint exception
         '4':'0':_ -> HTTP.status500 -- tx rollback
-        '5':'3':_ -> HTTP.status503 -- insufficient resources
+        '5':'3':_ -> HTTP.status500 -- insufficient resources
```

Comment:
How about matching `53400` exactly here for status500 and then leaving the other `53xxx` as 503?

As mentioned on https://github.com/PostgREST/postgrest/issues/3257#issuecomment-1962327567, the other `53xxx` do seem fit for status503.